web/impact/impact/permissions.py

```diff
@@ -39,17 +49,11 @@ def has_permission(self, request, view):
             name=settings.V1_CONFIDENTIAL_API_GROUP).exists()
 
 
-class DynamicModelPermissions(BasePermission):
-    perms_map = {
-        'GET': ['%(app)s.view_%(model_name)s'],
-        'OPTIONS': ['%(app)s.view_%(model_name)s'],
-        'HEAD': ['%(app)s.view_%(model_name)s'],
-        'POST': ['%(app)s.add_%(model_name)s'],
-        'PUT': ['%(app)s.change_%(model_name)s'],
-        'PATCH': ['%(app)s.change_%(model_name)s'],
-        'DELETE': ['%(app)s.delete_%(model_name)s'],
-    }
+def method_to_perm(method):
+    return ["%(app)s.{}_%(model_name)s".format(METHOD_TO_ACTION[method])]
```

Comment:
maybe we should do METHOD_TO_ACTION.get(method, 'get') and default to 'get'

Reply:
We could.  I didn't want to change the original behavior which was `self.perms_map[request.method]`.  I don't expect it actually matters, but it was my reasoning.  Don't have a strong opinion.  Happy to change it if you want since it should be a bit safer and I'd be surprised if anything in this code is depending on it raising an exception if the method is missing.

Reply:
Actually 'get' is not a good default since this is create a Django permission name.  For "get" the value is "view" so we could use that.  However, given that this is permission stuff, I think I'd rather throw an exception if we get something unexpected.  Won't fix (unless you make a really strong argument).